python/tox.ini

```diff
@@ -4,18 +4,22 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py27,py34,py35
+envlist = py27,py34,py35,py38,py39,py310,py311,py312
```

Comment:
Not an action for this PR, but we should probably officially drop 2.7 support. 

We might also want to drop support for any version that's [EoL for a decade] (so everything < 3.5)

Reply:
I'd be glad to add that in this PR(as a separate commit) if that's wanted

Reply:
No, let's keep that a separate PR for now, since it's a potential breaking change.